### PR TITLE
feat: esm

### PR DESCRIPTION
**Pull request by ThaUnknown**
**What is the purpose of this pull request? (put an "X" next to item)**

- [x] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Other, please explain:

**What changes did you make? (Give an overview)**

- chore: update docs and package description
- refactor: switch to esm imports and exports

**Which issue (if any) does this pull request address?**
- https://github.com/webtorrent/webtorrent/issues/2177

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
-const Tracker = require('bittorrent-tracker/client')
-const LSD = require('bittorrent-lsd')
+import debugFactory from 'debug'
+import DHT from 'bittorrent-dht/client.js' // empty object in browser
+import { EventEmitter } from 'events'
+import parallel from 'run-parallel'
+import Tracker from 'bittorrent-tracker/client.js'
+import LSD from 'bittorrent-lsd'
+
+const debug = debugFactory('torrent-discovery')
```
Comment (DiegoRBaquero) [inline index.js:9]:
```suggestion
import Debug from 'debug'
import DHT from 'bittorrent-dht/client.js' // empty object in browser
import { EventEmitter } from 'events'
import parallel from 'run-parallel'
import Tracker from 'bittorrent-tracker/client.js'
import LSD from 'bittorrent-lsd'

const debug = debug('torrent-discovery')
```

Comment (ThaUnknown) [reply index.js:9]: this is a perfect example of why I didn't want to do this, Debug is too close to debug and its very easy to mistake the 2 like you just did

Comment (DiegoRBaquero) [reply index.js:9]: True, but capital makes it obvious, just like with other imports

Comment (ThaUnknown) [reply index.js:9]: no other compromises here? @DiegoRBaquero

**Review by DiegoRBaquero — APPROVED** [review]

**Pull request merged**